db/channel_cache.go

```diff
@@ -83,11 +83,11 @@ type channelCacheImpl struct {
 }
 
 func NewChannelCacheForContext(terminator chan bool, options ChannelCacheOptions, context *DatabaseContext) (*channelCacheImpl, error) {
-	return newChannelCache(context.Name, terminator, options, context, context.activeChannels, context.DbStats.Cache())
+	return newChannelCache(context.Name, terminator, context.terminated, options, context, context.activeChannels, context.DbStats.Cache())
```

Comment:
The channel cache is being initialized by the changeCache.  If the changeCache is going to manage it's own teardown (in changeCache.Stop(), the channel cache should be adding it's background task to changeCache.terminator, not the database context terminator.

Reply:
Makes sense, this change has been included in the latest revision.